build-system/eslint-rules/no-non-string-log-args.js

```diff
@@ -0,0 +1,131 @@
+  {name: 'assertBoolean', variadic: false, startPos: 1},
+  {name: 'assertEnumValue', variadic: false, startPos: 2},
+  {name: 'assertElement', variadic: false, startPos: 1},
+  {name: 'createExpectedError', variadic: true, startPos: 0},
+  {name: 'fine', variadic: true, startPos: 1},
+  {name: 'info', variadic: true, startPos: 1},
+  {name: 'warn', variadic: true, startPos: 1},
+  {name: 'error', variadic: true, startPos: 1},
+  {name: 'expectedError', variadic: true, startPos: 1},
+  {name: 'createError', variadic: true, startPos: 0},
+];
+
+/**
+ * @param {!Node} node
+ * @return {boolean}
+ */
+function areAllArgumentsLiteral(node) {
+  if (node.type === 'Literal') {
+    return true;
+  }
+  // Allow for string concatenation operations.
+  if (node.type === 'BinaryExpression' && node.operator === '+') {
+    if (areAllArgumentsLiteral(node.left) &&
+        areAllArgumentsLiteral(node.right)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * @param {string} name
+ * @return {!LogMethodMetadata}
+ */
+function getMetadata(name) {
+  for (let i = 0; i < transformableMethods.length; i++) {
+    const curTransformableMethod = transformableMethods[i];
+    if (curTransformableMethod.name === name) {
+      return curTransformableMethod;
+    }
+  }
+  return null;
+}
+
+const expressions = transformableMethods.map(x => {
+  return `CallExpression[callee.property.name=${x.name}]`;
+}).join(',');
+
+
+module.exports = function(context) {
+  return {
+    [expressions]: function(node) {
+      // Make sure that callee is a CallExpression as well.
+      // dev().assert() // enforce rule
+      // dev.assert() // ignore
+      if (!(node.callee.object &&
+          node.callee.object.type === 'CallExpression')) {
+        return;
+      }
+
+      // Make sure that the CallExpression is one of dev() or user().
+      if(!['dev', 'user'].includes(node.callee.object.callee.name)) {
+        return;
+      }
+
+      const methodInvokedName = node.callee.property.name;
+      // Find the position of the argument we care about.
+      const metadata = getMetadata(methodInvokedName);
+
+      // If there's no metadata, this is most likely a test file running
+      // private methods on log.
+      if (!metadata) {
+        return;
+      }
+
+      const argToEval = node.arguments[metadata.startPos];
+
+      if (!argToEval) {
+        context.report({
+          node: node,
+       	  message: `No argument passed into expected message parameter ` +
+            `number ${metadata.startPos}`,
+        });
+        return;
+      }
+
+      if (!areAllArgumentsLiteral(argToEval)) {
+        context.report({
+          node: argToEval,
+       	  message: `Must use a literal string at method invocation ` +
+              `"${metadata.name}" on argument position ${metadata.startPos}. ` +
+              `No other Node type is allowed besides Raw strings or string ` +
+              `concatenation operations.`,
```

Comment:
Would it be possible to make this message more succinct? This is making the lint output very verbose, and messages are spilling into multiple lines.